### DISCUSSION TITLE
Samplerate with Camilla DSP : capture sample rate can't get over 44100khz

### DISCUSSION THREAD

**Discussion by NexusPenguin** [rern/rAudio#1288] [General]
Hi, 

If I turn on Camilla DSP, whatever I do, I will not get output sound that is above 44.1khz. 

Inoput is in 24/96 and is reported so. But as soon as I nable Camilla DSP I see "Capture samplerate : 44100" and my DAC reports that input is PCM 44 when it should be PCM 96. 

Not sure I'll notice a difference, but if I have files in 24/96, I must say that I would appreciate listening to those in 24/96. 

I'll try other Raspberry pi4 with Camilla DSP to see if they have the same "issue". 

Or maybe something is not set as it should in my system ? 

Because, this and "Lyrics not available" that no one experience except me, makes me believe something is wrong somewhere. 

Regards.

**Comment by NexusPenguin** [comment]
Really ?

No one with the same problem ?

**Reply by bnilsson11** [reply]
Yes, sort of.
The config file /srv/http/data/camilladsp/configs/camilladsp.yml has a setting for the sample rate, currently 44100.
I guess if you pass HD music through the DSP it is downsampled to 44100.
If I change the sample rate in the file it still plays fine, but the DSP function is lost.
Nothing happens when I use the controls.

And another disturbing this is that if I add another config file, e.g. camilladsp2.yml, it appears as a selection in the DSP page, but if I select camilladsp2.yml the DSP crashes. 
The reason is that the file /etc/default/camilladsp contains the following:
`ADDRESS=0.0.0.0
CONFIG=/srv/http/data/camilladsp/configs/.yml
LOGFILE=/var/log/camilladsp.log
MUTE=
PORT=1234
GAIN=-g0
` 
so the name reference to the config file in the CONFIG line is lost. Hence the crash.
Something for the developers to fix!

Will the DSP support higher sample rates in the future?

rAudio 20231111
6.1.28-5 aarch64
Raspberry Pi 4B Rev 1.5
BCM2711 • 7.6 GiB
4 x Cortex-A72 @ 1.50 GHz

BN

**Reply by bnilsson11** [reply]
Another thing with the DSP:
If a new filter is added, it is not functional.
Below is a cutout of the camilladsp.yml config file:
`filters:
  Bass:
    type: Biquad
    parameters:
      type: Lowshelf
      freq: 85.0
      q: 0.9
      gain: -6.0
  Treble:
    type: Biquad
    parameters:
      type: Highshelf
      freq: 6500.0
      q: 0.7
      gain: 0.0
  Volume:
    type: Volume
    parameters:
      ramp_time: 200.0
pipeline:
  - type: Filter
    channel: 0
    names:
      - Volume
      - Bass
      - Treble
  - type: Filter
    channel: 1
    names:
      - Volume
      - Bass
      - Treble
`

If I no add a new filter from the webGUI "+" function, such as "MidPeak", Biquad, Peaking, 800Hz, Q 1.414 the config file looks like this:

`filters:
  Volume:
    type: Volume
    parameters:
      ramp_time: 200.0
  Treble:
    type: Biquad
    parameters:
      type: Highshelf
      freq: 6500.0
      q: 0.7
      gain: 0.0
  MidPeak:
    type: Biquad
    parameters:
      type: Peaking
      freq: 1000.0
      q: 1.414
      gain: 0.0
  Bass:
    type: Biquad
    parameters:
      type: Lowshelf
      freq: 85.0
      q: 0.9
      gain: -6.0
pipeline:
  - type: Filter
    channel: 0
    names:
      - Volume
      - Bass
      - Treble
  - type: Filter
    channel: 1
    names:
      - Volume
      - Bass
      - Treble
`

The new filter shows up on the DSP web page, but it is not functional.
I am absolutely no expert in the camilladsp product, but I suspect the new name should also appear in the "pipeline" section.
If I manually edit the config file and add the filter name "MidPeak" in pipeline channel 0 and channel 1 names, and restart the camilladsp.service, it starts working.

.

**Reply by bnilsson11** [reply]
An update on the camilladsp.
Today, strangely enough, 192000 sr was working ok on the DSP, after I changed it in the config file.
I could hear the effect on the music, and my connected DAC indicated 192k on the display.
Not sure what was going on before...

**Reply by rern** [reply]
With update 20231118, try adding the new filter to existing Pipelines. 
Click flowchart, the icon before plus icon, to see it in signal diagram.

**Comment by bnilsson11** [comment]
Ok, got it.
Two steps are needed, crate the filter and then add it to the pipeline?
How can I remove a filter? I understand I can remove it from the pipeline, but how to remove the filter itself?

I tried to change the config from my roomeq.yml to camilladsp.yml by the dropdown menu.
Not working.
When I do that the /etc/default/camilladsp becomes:

ADDRESS=0.0.0.0
CONFIG=/srv/http/data/camilladsp/configs/roomeq.yml.yml
LOGFILE=/var/log/camilladsp.log
MUTE=
PORT=1234
GAIN=-g0



> 19 nov. 2023 kl. 05:13 skrev rern ***@***.***>:
> 
> 
> With update 20231118, try adding the new filter to existing Pipelines.
> Click flowchart, the icon before plus icon, to see it in signal diagram.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/rern/rAudio/discussions/1288#discussioncomment-7609360>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAZXLDSQFKMVSS4V3KSNSJTYFGBQDAVCNFSM6AAAAAAWX5NCP6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMMBZGM3DA>.
> You are receiving this because you commented.
> 

Bengt Nilsson
Ansåsvägen 8
43853 Hindås
Sweden
***@***.***

**Comment by bnilsson11** [comment]
Also, DSP Volume is not working. Should it?


> 19 nov. 2023 kl. 05:13 skrev rern ***@***.***>:
> 
> 
> With update 20231118, try adding the new filter to existing Pipelines.
> Click flowchart, the icon before plus icon, to see it in signal diagram.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/rern/rAudio/discussions/1288#discussioncomment-7609360>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAZXLDSQFKMVSS4V3KSNSJTYFGBQDAVCNFSM6AAAAAAWX5NCP6VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TMMBZGM3DA>.
> You are receiving this because you commented.
> 

Bengt Nilsson
Ansåsvägen 8
43853 Hindås
Sweden
***@***.***
